source/poodinis/container.d

```diff
@@ -192,6 +201,44 @@ synchronized class DependencyContainer {
         return newRegistration;
     }
 
+    /**
+     * 
+     */
+    Registration register(SuperType, ConcreteType : SuperType)(InjectionInitializer!ConcreteType creator, 
+            RegistrationOption options = RegistrationOption.none) if (!is(ConcreteType == struct)) {
+
+        TypeInfo registeredType = typeid(SuperType);
+        TypeInfo_Class concreteType = typeid(ConcreteType);
+
+        debug(poodinisVerbose) {
+            writeln(format("DEBUG: Register type %s (as %s)", concreteType.toString(), registeredType.toString()));
+        }
+
+        auto existingRegistration = getExistingRegistration(registeredType, concreteType);
+        if (existingRegistration) {
+            return existingRegistration;
+        }
+
+        InstanceFactory instanceFactory = new class InstanceFactory {
+            protected override Object createInstance() { 
+                return creator();
+            }
+        };
+
+        auto newRegistration = new AutowiredRegistration!ConcreteType(registeredType, instanceFactory, this);
+        newRegistration.singleInstance();
+
+        static if (!is(SuperType == ConcreteType)) {
+            if (!hasOption(options, persistentRegistrationOptions, RegistrationOption.doNotAddConcreteTypeRegistration)) {
+                auto concreteTypeRegistration = register!ConcreteType;
+                concreteTypeRegistration.linkTo(newRegistration);
+            }
+        }
+
+        registrations[registeredType] ~= cast(shared(Registration)) newRegistration;
+        return newRegistration;
+    }
```

Comment:
This function is now duplicating a lot of code from
public Registration register(SuperType, ConcreteType : SuperType)(
            RegistrationOption options = RegistrationOption.none) if (!is(ConcreteType == struct)) 

Is there a way to call this function in that one?